Improve spacing between quantity table columns

jpetrell opened the pull request:
Just small adjustment to the column sizes to make sure there are enough margins around the wider energy (kWh) column. Fixes #479.

![image](https://github.com/victronenergy/gui-v2/assets/2203667/85dd5370-eefd-4a80-9c3d-07616f312361)

chriadam commented:
In 5-inch mode, this can cause the text to elide for solar chargers.  The text is "External control" and it renders properly without the patch applied.
<img width="600" alt="venus-elide-text-issue" src="https://github.com/victronenergy/gui-v2/assets/2569776/aaadf4d7-c810-460d-961c-4c05d9ca3dce">


The width of the title label in QuantityTable.qml is set via `width: root.availableWidth - headerQuantityRow.width`  so I guess the patch causes the headerQuantityRow.width to increase in some cases?

jpetrell commented:
> The text is "External control" and it renders properly without the patch applied

That is actually unrelated, my changes actually reduced the total quantity row width a bit. But not enough, "External control" was still truncated. I fixed it now, also ended up refactoring the column width calculations to own component to avoid code duplication. Dropped the magical multipliers and width boost values, instead measuring now the maximum column size using FontMetrics (+ some spacing between the columns). Adjusted the seven inch variant to have more spacing so the layouts is still more evenly spread around the available space.

![image](https://github.com/victronenergy/gui-v2/assets/2203667/28413d5a-4b5b-44f1-bc9a-d0f340526f24)

blammit commented on components/QuantityTableMetrics.qml:
```diff
@@ -0,0 +1,40 @@
+
+		// Give the unit symbol some extra space on the column.
+		const maxTextWidth = unit === VenusOS.Units_Energy_KiloWattHour ? advanceWidth("9999kWH")
+																		: advanceWidth("9999W")
```
Odd indentation? Maybe it's fine to just have it all on the same line

jpetrell replied:
Fixed.

blammit reviewed (approved):
Looks great! The solution is a bit more complicated than I expected, but I think it's worthwhile to use the metrics to calculate things properly instead of just putting in random values and hoping it will work.

jpetrell commented:
> The solution is a bit more complicated than I expected

It may look daunting, but it fixes multiple issues and mostly it is just moving related properties and _quantityColumnWidth() function to a shared component to avoid code duplication between QuantityTable.qml and QuantityTableSummary.qml.

jpetrell commented:
> The solution is a bit more complicated than I expected

Also, the PR started with just small adjustments, but then Chris noticed other issues which could no longer be fixed by simply tweaking the multipliers. Also I was worried about the translations, any issue I fix with the magical values will be English-specific fix if I don't take the actual text dimensions into account.
